File: php/features.php
<?php
declare(strict_types=1);

// AiBackgroundRemover SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class AiBackgroundRemoverFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new AiBackgroundRemoverBaseFeature();
            case "test":
                return new AiBackgroundRemoverTestFeature();
            default:
                return new AiBackgroundRemoverBaseFeature();
        }
    }
}

File: py/features.py
# AiBackgroundRemover SDK feature factory

from feature.base_feature import AiBackgroundRemoverBaseFeature
from feature.test_feature import AiBackgroundRemoverTestFeature


def _make_feature(name):
    features = {
        "base": lambda: AiBackgroundRemoverBaseFeature(),
        "test": lambda: AiBackgroundRemoverTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
